pkg/sql/logictest/testdata/logic_test/alter_column_type

```diff
@@ -161,3 +161,86 @@ ALTER TABLE t ALTER s TYPE STRING
 
 statement ok
 DROP TABLE t
+
```

Comment:
Add some tests for how this interacts with preceeding schema changes. Concurrent schema changes probably needs a schema_changer_test